compiler/luci-interpreter/src/kernels/Add.test.cpp

```diff
@@ -27,59 +27,118 @@ namespace
+  std::pair<float, int32_t> input2_quant_param =
+      quantizationParams<T>(getMinValue(input2_data), getMaxValue(input2_data));
+  Tensor input1_tensor{getElementType<T>(),
+                       input1_shape,
+                       {{input1_quant_param.first}, {input1_quant_param.second}},
+                       ""};
+  Tensor input2_tensor{getElementType<T>(),
+                       input2_shape,
+                       {{input2_quant_param.first}, {input2_quant_param.second}},
+                       ""};
+  if (std::is_floating_point<T>::value)
+  {
+    input1_tensor.writeData(input1_data.begin(), input1_data.size() * sizeof(T));
+    input2_tensor.writeData(input2_data.begin(), input2_data.size() * sizeof(T));
   }
-  // Re-run with exchanged inputs.
-  for (size_t i = 0; i < test_shapes.size(); ++i)
+  else
   {
-    Tensor input1_tensor = makeInputTensor<DataType::FLOAT32>(test_shapes[i], input2_data);
-    Tensor input2_tensor = makeInputTensor<DataType::FLOAT32>(base_shape, input1_data);
-    Tensor output_tensor = makeOutputTensor(DataType::FLOAT32);
+    std::vector<T> quantized_input1_value =
+        quantize<T>(input1_data, input1_quant_param.first, input1_quant_param.second);
+    std::vector<T> quantized_input2_value =
+        quantize<T>(input2_data, input2_quant_param.first, input2_quant_param.second);
+    input1_tensor.writeData(quantized_input1_value.data(),
+                            quantized_input1_value.size() * sizeof(T));
+    input2_tensor.writeData(quantized_input2_value.data(),
+                            quantized_input2_value.size() * sizeof(T));
+  }
+  std::pair<float, int32_t> output_quant_param =
+      quantizationParams<T>(getMinValue(input1_data) + getMinValue(input2_data),
+                            getMaxValue(input1_data) + getMaxValue(input2_data));
+  Tensor output_tensor =
+      makeOutputTensor(getElementType<T>(), output_quant_param.first, output_quant_param.second);
 
-    AddParams params{};
-    params.activation = Activation::RELU;
+  AddParams params{};
+  params.activation = Activation::RELU;
 
-    Add kernel(&input1_tensor, &input2_tensor, &output_tensor, params);
-    kernel.configure();
-    kernel.execute();
+  Add kernel(&input1_tensor, &input2_tensor, &output_tensor, params);
+  kernel.configure();
+  kernel.execute();
 
-    EXPECT_THAT(extractTensorData<float>(output_tensor),
-                ::testing::ElementsAreArray(ArrayFloatNear(test_outputs[i], 0.0001f)))
-        << "With shape number " << i;
+  if (std::is_floating_point<T>::value)
+  {
+    EXPECT_THAT(extractTensorData<T>(output_tensor), ElementsAreArray(ArrayFloatNear(output_data)));
+  }
+  else
+  {
+    EXPECT_THAT(dequantize<T>(extractTensorData<T>(output_tensor), output_tensor.scale(),
+                              output_tensor.zero_point()),
+                ElementsAreArray(ArrayFloatNear(output_data, kQuantizedTolerance)));
   }
```

Comment:
Can you add tests with exchanged inputs as the previous code did?

Reply:
Okay, I will add it.